### ISSUE TITLE
no-unreachable incorrectly flagging code as unreachable

### ISSUE THREAD

**Issue by jamorales-bsft** [eslint/eslint#6928]
**What version of ESLint are you using?**
v3.3.1
**What parser (default, Babel-ESLint, etc.) are you using?**
default
**Please show your full configuration:**
```
{
    "env": {
        "node": true,
        "mocha": true,
        "es6": true
    },
    "extends": "eslint:recommended",
    "rules": {
        "indent": "off",
	"no-irregular-whitespace": "off",
        "linebreak-style": [
            "error",
            "unix"
        ],
        "quotes": [
            "error",
            "double"
        ],
        "semi": [
            "error",
            "never"
        ]
    }
}
```
**What did you do? Please include the actual source code causing the issue.**
```
function somefunc () {
    if (isThisValueTrue) {
        return
    }
    
    console.log("this is reachable if isThisValueTrue is false")
    return
}
```
**What did you expect to happen?**
no error should be triggered
**What actually happened? Please include the actual, raw output from ESLint.**
an error was triggered

**Comment by ilyavolodin** [comment]
Could you provide an exact error message you are getting?

**Comment by jamorales-bsft** [comment]
43:3   error  Unreachable code                     no-unreachable

**Comment by mysticatea** [comment]
I cannot reproduce this on [our online demo](http://eslint.org/demo/).

![image](https://cloud.githubusercontent.com/assets/1937871/17764178/b29c732e-6558-11e6-9997-e841d9cc76d0.png)

**Comment by jamorales-bsft** [comment]
ok it is my error there is a throw above the return. so it is the return itself that is unreachable

**Comment by ilyavolodin** [comment]
Closing, as the issue is resolved.
